app/services/discord.server.ts

```diff
@@ -1,23 +1,60 @@
-import type { DiscordGuildMember } from "~/models/discordProfile.server";
+import redisClient from "~/services/redis.server";
 
 export async function getDiscordGuildMembershipByProfileId(
   id: string
-): Promise<DiscordGuildMember | null> {
+): Promise<boolean> {
   let discordBotToken = process.env.DISCORD_BOT_TOKEN;
 
   const authHeaders = {
     Authorization: `Bot ${discordBotToken}`,
   };
 
-  let discordGuildMember = await fetch(
-    `https://discordapp.com/api/guilds/${process.env.DISCORD_GUILD_ID}/members/${id}`,
-    {
-      headers: authHeaders,
+  async function getMoreGuildMembers(
```

Comment:
in my opinion this function is overly responsible and I see no reason for it to be nested inside another function.

I would create the following functions for a better seperation of concerns:
- f1: A function which fetches all guild members from the discord api directly
- f2: A function which returns all guild members by checking if in cache and returning that or by calling f1 and saving this data to redis and returning the response
- f3: The replacement for this parent function which would call f2 and do the lookup if user exists

Reply:
I've made this change!